kernel/thread.c

```diff
@@ -413,14 +413,15 @@ static char *setup_thread_stack(struct k_thread *new_thread,
 		stack_buf_start = K_KERNEL_STACK_BUFFER(stack);
 		stack_buf_size = stack_obj_size - K_KERNEL_STACK_RESERVED;
 
+#ifdef ARCH_KERNEL_STACK_RESERVED
```

Comment:
I don't think that `#ifdef ARCH_KERNEL_STACK_RESERVED` is the right approach. Some architectures appear to define  ARCH_KERNEL_STACK_RESERVED to 0, which in turn can lead to K_KERNEL_STACK_RESERVED being defined to 0.

Is there any reason why we the following could not be done to prevent that warning?

```
#if (K_KERNEL_STACK_RESERVED > 0)
...
#endif
```

Reply:
that works as well